dbms/src/Storages/DeltaMerge/Index/InvertedIndex/CommonUtil.cpp

```diff
@@ -0,0 +1,189 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <IO/ReadHelpers.h>
+#include <IO/WriteHelpers.h>
+#include <Storages/DeltaMerge/Index/InvertedIndex/CommonUtil.h>
+
+
+namespace DB::DM::InvertedIndex
+{
+
+template <typename T>
+void Block<T>::serialize(WriteBuffer & write_buf) const
+{
+    writeIntBinary(static_cast<UInt32>(entries.size()), write_buf);
+    // write all values first
+    for (const auto & entry : entries)
+    {
+        writeIntBinary(entry.value, write_buf);
+        writeIntBinary(static_cast<UInt32>(entry.row_ids.size()), write_buf);
+    }
+    // write all row_ids
+    for (const auto & entry : entries)
+    {
+        auto row_ids = entry.row_ids;
+        write_buf.write(reinterpret_cast<const char *>(row_ids.data()), row_ids.size() * sizeof(RowID));
+    }
+}
+
+template <typename T>
+void Block<T>::deserialize(Block<T> & block, ReadBuffer & read_buf)
+{
+    UInt32 size;
+    readIntBinary(size, read_buf);
+    block.entries.resize(size);
+    for (UInt32 i = 0; i < size; ++i)
+    {
+        T value;
+        readIntBinary(value, read_buf);
+        UInt32 row_ids_size;
+        readIntBinary(row_ids_size, read_buf);
+        block.entries[i].value = value;
+        block.entries[i].row_ids.resize(row_ids_size);
+    }
+    for (UInt32 i = 0; i < size; ++i)
+    {
+        auto & entry = block.entries[i];
+        read_buf.read(reinterpret_cast<char *>(entry.row_ids.data()), entry.row_ids.size() * sizeof(RowID));
+    }
```

Comment:
I am not sure if it is good handle some EOF failures here? Because I can only expect a data corruption happens here.